src/main/kotlin/nl/info/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -356,14 +358,22 @@ class ProductaanvraagService @Inject constructor(
         rolType: RolType,
         vestigingsNummer: String,
         zaak: URI
-    ) = zrcClientService.createRol(
-        RolVestiging(
-            zaak,
-            rolType,
-            ROL_TOELICHTING,
-            VestigingIdentificatie().apply { this.vestigingsNummer = vestigingsNummer }
+    ): Rol<*> {
+        if (!vestigingsNummer.isValidKvkVestigingsnummer()) {
+            throw ProductaanvraagNotSupportedException(
+                "Invalid KVK vestigingsnummer: '$vestigingsNummer'. " +
+                    "It should be a 12-digit number."
```

Comment:
Might be too much, but shouldn't `12` be taken from `KVK_VESTIGINGSNUMMER_LENGTH` ?

Reply:
I personally find it better never to use our own main code in our tests. I think our tests should treat our main code as a black box (more or less) in that sense. So I actually think it is better not to use `KVK_VESTIGINGSNUMMER_LENGTH` in our tests. Because if, by accident, someone ever changes this constant in our main code, this unit test should fail.